### PR TITLE
add scalevar theoryids for n3lo+qed mhou covmat

### PR DESCRIPTION
**Pull request by RoyStegeman**
The PDF of https://arxiv.org/abs/2406.01779 was done with a hack where the covmat was not generated during vp-setupfit but instead copied from a pure QCD fit

### CODE REVIEW

File: validphys2/src/validphys/scalevariations/scalevariationtheoryids.yaml [hunk 1]
```diff
@@ -218,6 +218,86 @@ scale_variations_for:
       (1, 0.5 hadronic): 796  # As 787 but DIS from 722
       (1, 2 hadronic): 797 # As 790 but DIS from 722
 
+  - theoryid: 824 # N3LO+QED
```
[the remark this answers is not in the document]
Comment (RoyStegeman) [reply validphys2/src/validphys/scalevariations/scalevariationtheoryids.yaml:221]: No it's calculated using the list of variations declared below this line (combined with the list in pointprescriptions.yaml). Then in construction.py it should be defined which in the list of theories is the central one

[the remark this answers is not in the document]
Comment (giacomomagni) [reply validphys2/src/validphys/scalevariations/scalevariationtheoryids.yaml:221]: okay, yes you're right.

**Comment by RoyStegeman** [comment]
Comparison to the N3LO+QED one done by AB and NL: https://vp.nnpdf.science/B6HAP6oWTjmoWte51oBJNw==/

For some reason I couldn't do the full vp-comparefits report because it complained about different size covmats? Did the division in terms of processes change since the fits for the N3LO paper (since AB and NL used that covmat instead of recomputing their own, hence the need for this PR)?

**Comment by giacomomagni** [comment]
Nice the two fits are in perfect agreement. 
Datasets division should be the same with the exception that we used the old commondata, so something might have changed?

**Comment by RoyStegeman** [comment]
Good point, probably the covmats have different names for the same dataset and while the dataset_names.yml mapping is applied to loading of the datasets when doing a fit, the different names may still create a problem when dealing with the covmats.

**Comment by scarlehoff** [comment]
> the different names may still create a problem when dealing with the covmats.

If they do, that's a bug _somewhere_. In particular the size of the covmat should be the same.

It might be that loading a covmat from file should also go through the mapping though. Roughly here https://github.com/NNPDF/nnpdf/blob/a89d1c7a25e1c8dfb791b28c14b46882a482e069/validphys2/src/validphys/config.py#L786 the names on the dataframe (one of the levels of the dataframe, don't ask me which, pandas makes my head spin) need some variation of `<something pandas>.map(lambda x: legacy_to_new_map(x, None))`

**Comment by RoyStegeman** [comment]
> If they do, that's a bug somewhere. In particular the size of the covmat should be the same.

One is always (0,0) so even if the size is the same, it simply doesn't get any elements in one of the matrices with the correct label.

**Comment by scarlehoff** [comment]
Yes, that make sense. Changing the names of the dataframe in the loader function I linked should hopefully be enough.

**Review by giacomomagni — APPROVED** [review]

**Comment by giacomomagni** [comment]
unless you don't want to fix the th covmat compatibility issue  here I'm happy.

**Comment by scarlehoff** [comment]
Same. Open an issue though, it would be nice to have it fixed.

**Pull request merged**